entities/src/main/java/org/odk/collect/entities/javarosa/filter/LocalEntitiesFilterStrategy.kt

```diff
@@ -87,7 +87,13 @@ class LocalEntitiesFilterStrategy(entitiesRepository: EntitiesRepository) :
 
         return if (candidate != null) {
             val child = candidate.nodeSide.steps[0].name.name
-            val value = candidate.evalContextSide(sourceInstance, evaluationContext) as String
+            var value = candidate.evalContextSide(sourceInstance, evaluationContext)
+
+            value = if (value is Double && value % 1 == 0.0) {
+                value.toInt().toString()
```

Comment:
I don't believe this would work correctly if the expression was `age = 25.0` and the value of age was `"25.0"`.